### PR TITLE
[master] PLT-302 Optimize Persona ES alias

### PR DESCRIPTION
**Pull request by nikhilbonte21**

### CODE REVIEW

**Review by checkaayush — APPROVED** [review]
LGTM

File: repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java [hunk 1]
```diff
@@ -165,32 +170,50 @@ private Map<String, Object> getFilterForPurpose(AtlasEntity purpose) throws Atla
...
+        List<String> terms = new ArrayList<>();
         for (AtlasEntity policy: policies) {
 
             if (policy.getStatus() == null || AtlasEntity.Status.ACTIVE.equals(policy.getStatus())) {
                 List<String> assets = getPolicyAssets(policy);
 
                 if (getIsAllowPolicy(policy)) {
                     if (getPolicyActions(policy).contains(ACCESS_READ_PERSONA_METADATA)) {
+
+                        int assetSize = terms.size() + assets.size() + 1;
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java:182]: Why are we using terms.size here? Also, "terms" array will be empty according to the code flow when this check will be evaluated.

Comment (checkaayush) [reply repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java]: The asset max limit check has been simplified and the changes should resolve questions here. Let me know if that isn't the case.

**Comment by nikhilbonte21** [comment]
Changes made by @checkaayush look good to me. We can merge this once @mehtaanshul reviews once again

**Review by mehtaanshul — APPROVED** [review]

**Pull request merged**